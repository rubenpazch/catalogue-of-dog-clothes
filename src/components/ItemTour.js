import React from 'react';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Container from 'react-bootstrap/Container';
import styles from '../css/itemtour.module.css';

const ItemTour = ({ image, text }) => (
  <Row className={styles.wrapper}>
    <img src={image} />
    <p>{text}</p>
  </Row>
);

export default ItemTour;
