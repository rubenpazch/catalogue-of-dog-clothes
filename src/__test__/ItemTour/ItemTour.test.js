import React from 'react';
import ReactDOM from 'react-dom';
import {
  BrowserRouter as Router,
  Route,
} from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ItemTour from '../../components/ItemTour';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

it('renders ItemTours without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Router>
      <Route>
        <ItemTour titleTour="Title" imgPath="path" amount="12" id="8" />
      </Route>
    </Router>,
    div,
  );
});

it('renders item correctly', () => {
  const { getByTestId } = render(
    <Router>
      <Route>
        <ItemTour titleTour="Title" imgPath="path" amount="12" id="8" />
      </Route>
    </Router>,
  );
  getByTestId('8');
});

it('render button information', () => {
  const { getByTestId } = render(
    <Router>
      <Route>
        <ItemTour titleTour="Title" imgPath="path" amount="12" id="8" />
      </Route>
    </Router>,
  );
  expect(getByTestId('8')).toHaveTextContent('Title');
  expect(getByTestId('8')).toHaveTextContent('12');
});

it('machets snapshot itemtour', () => {
  const itemTour = renderer.create(
    <Router>
      <Route>
        <ItemTour titleTour="Title" imgPath="path" amount="12" id="8" />
      </Route>
    </Router>,
  ).toJSON();
  expect(itemTour).toMatchSnapshot();
});
