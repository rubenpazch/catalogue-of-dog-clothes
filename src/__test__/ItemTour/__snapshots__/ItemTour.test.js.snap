// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`machets snapshot itemtour 1`] = `
<div
  className="wrapper col"
  data-testid="8"
>
  <img
    alt="tour1"
    src="path"
  />
  <div
    className="header"
  >
    <h5
      md={9}
    >
      <a
        href="/tourdetail/8"
        onClick={[Function]}
      >
        Title
      </a>
    </h5>
    <span
      md={3}
    >
      12
    </span>
  </div>
  <div
    className="services"
  >
    <ul
      className="listServices"
    >
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-coffee fa-w-20 "
          data-icon="coffee"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Guiding
      </li>
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-coffee fa-w-20 "
          data-icon="coffee"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Transportation
      </li>
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-coffee fa-w-20 "
          data-icon="coffee"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Entrances
      </li>
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-coffee fa-w-20 "
          data-icon="coffee"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        English
      </li>
    </ul>
    <div>
      text
    </div>
  </div>
</div>
`;
